pkg/provider/consulcatalog/config.go

```diff
@@ -300,3 +293,13 @@ func (p *Provider) addServer(ctx context.Context, item itemData, loadBalancer *d
 func itemServersTransportKey(item itemData) string {
 	return provider.Normalize("tls-" + item.Namespace + "-" + item.Datacenter + "-" + item.Name)
 }
+
+func getName(i itemData) string {
+	if !i.ExtraConf.ConsulCatalog.Canary {
+		return provider.Normalize(i.Name)
+	}
+
+	hasher := fnv.New64()
+	hasher.Write([]byte(strings.Join(i.Tags, ",")))
```

Comment:
I am going out on a limb here and wonder if it would make sense to sort the tags before joining. I am not sure if the tag order is guranteed.

Reply:
Thanks for the feedback,

The hash has no other purpose than to produce a unique name temporarily, so we didn't sort the tags on purpose because it doesn’t really matter here (if we didn’t miss something).

Reply:
I guess it all depends on whether the tags returned by the consul catalog are guaranteed to keep order. Otherwise you will probably see a config change every X seconds (X = refreshinterval) even if it basically results in a noop. In that sense using sort is just a means of better stability against external changes -- I agree that it might not matter, but given that the tag list is probably small enough it won't hurt that much either. YMMV